### PR TITLE
feat: Gradle use datasource

### PR DESCRIPTION
**Pull request by corecanarias**
Closes #

### CODE REVIEW

File: lib/manager/gradle/index.js [hunk 1]
```diff
@@ -25,122 +25,53 @@ async function extractAllPackageFiles(config, packageFiles) {
...
-        gradleFiles.push({
-          packageFile,
-          manager: 'gradle',
-          ...deps,
-        });
-      }
+      gradleFiles.push({
+        packageFile,
+        manager: 'gradle',
+        datasource: 'maven',
```
Comment (corecanarias) [inline lib/manager/gradle/index.js]: @rarkins I think last time I missed the datasource :?

Comment (rarkins) [reply lib/manager/gradle/index.js]: Yeah, but I think I fixed that the first time too, so it maybe doesn't explain the problem

**Comment by rarkins** [comment]
@ChristianMurphy do you have the time to test out this branch on your use case? I want to merge it in the next 1-2 days as it's holding up another PR

**Comment by ChristianMurphy** [comment]
Tried this out by checking out this branch, and using `yarn start -- --token {token} Jasig/uPortal`.

It did open two PRs, a good sign :+1:
https://github.com/Jasig/uPortal/pull/1527
https://github.com/Jasig/uPortal/pull/1528

There is some new behavior that I'm not sure if it is expected or not.
One new behavior, is an uptick of logs about `javax.ccpp:ccpp`

```
 INFO: No versions found for javax.ccpp:ccpp in 1 repositories (repository=Jasig/uPortal)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=settings.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/uPortal-api-internal/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/uPortal-api-platform-impl/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/uPortal-api-platform/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/uPortal-api-rest/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-api/uPortal-api-search/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-concurrency/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-content/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-content/uPortal-content-portlet/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-content/uPortal-content-publishing/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-events/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-filesystem/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-grouper/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-ldap/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-local/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-pags/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-groups/uPortal-groups-smartldap/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-health/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-hibernate/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-hibernate/uPortal-hibernate3-dialects/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-hibernate/uPortal-hibernate4-dialects/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-i18n/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-io/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-io/uPortal-io-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-io/uPortal-io-jaxb/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-io/uPortal-io-types/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-layout/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-layout/uPortal-layout-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-layout/uPortal-layout-impl/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-marketplace/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-persondir/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-portlets/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-rdbm/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-rendering/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-authn/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-mvc/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-permissions/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-services/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-security/uPortal-security-xslt/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-soffit/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-soffit/uPortal-soffit-connector/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-soffit/uPortal-soffit-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-soffit/uPortal-soffit-renderer/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-spring/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-tenants/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-tools/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-url/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-utils/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-utils/uPortal-utils-core/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-utils/uPortal-utils-jmx/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-web/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=uPortal-webapp/build.gradle, dependency=javax.ccpp:ccpp)
 INFO: Failed to look up dependency javax.ccpp:ccpp (repository=Jasig/uPortal, packageFile=gradle.properties, dependency=javax.ccpp:ccpp)
```

Another is that some PRs were autoclosed (log below).
`com.google.guava:guava to 27.0.1-jre` and `org.jgroups:jgroups to 4.0.17.final` may be the new maven parser taking affect. :thinking: 
`com.sun.xml.bind:jaxb-impl to 2.3.2` and `com.sun.xml.bind:jaxb-xjc to 2.3.2` do seem to be valid semver/maven versions. :man_shrugging: 

```
INFO: PR autoclosed (repository=Jasig/uPortal)
       "prNo": 1519,
       "prTitle": "chore(deps): update dependency com.google.guava:guava to 27.0.1-jre"
 INFO: Deleting orphan branch (repository=Jasig/uPortal, branch=renovate/com.google.guava-guava-.x)
 INFO: PR autoclosed (repository=Jasig/uPortal)
       "prNo": 1520,
       "prTitle": "chore(deps): update dependency com.sun.xml.bind:jaxb-impl to 2.3.2"
 INFO: Deleting orphan branch (repository=Jasig/uPortal, branch=renovate/com.sun.xml.bind-jaxb-impl-.x)
 INFO: PR autoclosed (repository=Jasig/uPortal)
       "prNo": 1521,
       "prTitle": "chore(deps): update dependency com.sun.xml.bind:jaxb-xjc to 2.3.2"
 INFO: Deleting orphan branch (repository=Jasig/uPortal, branch=renovate/com.sun.xml.bind-jaxb-xjc-.x)
 INFO: PR autoclosed (repository=Jasig/uPortal)
       "prNo": 1523,
       "prTitle": "chore(deps): update dependency org.jgroups:jgroups to 4.0.17.final"
 INFO: Deleting orphan branch (repository=Jasig/uPortal, branch=renovate/org.jgroups-jgroups-.x)
 INFO: Request stats (repository=Jasig/uPortal)
       "rateLimit": 5000,
       "requestCount": 124,
       "rateLimitRemaining": 4867
```

**Comment by ChristianMurphy** [comment]
Also tested on `Jasig/NotificationPortlet`, PR rebasing and PR opening appears to be working as expected on this repo :ballot_box_with_check: :+1:

**Comment by rarkins** [comment]
For the autoclosed ones, this can be either because our new logic produces different lookup results *or* just because the branch names are now different (correct?!). The old ones look wrong. 

Also did you get only 2 because of hourly PR limiting?

**Comment by ChristianMurphy** [comment]
> For the autoclosed ones, this can be either because our new logic produces different lookup results _or_ just because the branch names are now different

Good to know.

> Also did you get only 2 because of hourly PR limiting?

It appears so, after that point the remaining logged as

```
INFO: Reached PR creation limit
```

**Comment by rarkins** [comment]
@corecanarias 	do you know if all the `Failed to look up dependency` messages indicate a bug, or just overzealous logging?

**Comment by corecanarias** [comment]
It seems that that dependency does not have a maven metadata file:
http://central.maven.org/maven2/javax/ccpp/ccpp/

Compared with this one, for example: http://central.maven.org/maven2/com/sun/xml/bind/jaxb-libs/

Not sure why it doesn't have the maven-metadata file. Maybe they uploaded it manually, somehow? I guess the best way to solve this is fixing the dependency itself

**Comment by corecanarias** [comment]
i've found other examples of libraries without maven-metadata file. All of them have only one dependency, maybe the file is generated with the second version?

In any case, for our current case it's not an issue

**Comment by rarkins** [comment]
Ready to merge? :)

**Comment by corecanarias** [comment]
It's a go for me ;)

**Comment by rarkins** [comment]
:boom:

**Pull request merged**